protocols/request-response/tests/ping.rs

```diff
@@ -201,6 +201,62 @@ fn emits_inbound_connection_closed_failure() {
     });
 }
 
+#[test]
+fn emits_inbound_connection_closed_if_channel_is_dropped() {
```

Comment:
```suggestion
/// We expect the substream to be properly closed when response channel is dropped.
/// Since the ping protocol used here expects a response, the sender considers this
/// early close as a protocol violation which results in the connection being closed.
/// If the substream were not properly closed when dropped, the sender would instead
/// run into a timeout waiting for the response.
#[test]
fn emits_inbound_connection_closed_if_channel_is_dropped() {
```
Just to clarify, since it can be misleading to think that it is the dropping of the response channel itself that results in the connection being closed. It is rather the protocol violation (received substream close instead of response) observed by the sender of the request.